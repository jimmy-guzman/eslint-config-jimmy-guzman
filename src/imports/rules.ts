import type { EslintRules } from '../models'

export const rules: EslintRules = {
  'import/consistent-type-specifier-style': 'off', // revisit
  'import/default': 'error',
  'import/dynamic-import-chunkname': 'off',
  'import/export': 'error',
  'import/exports-last': 'error',
  'import/extensions': [
    'error',
    'ignorePackages',
    {
      js: 'never',
      jsx: 'never',
      mjs: 'never',
      ts: 'never',
      tsx: 'never',
    },
  ],
  'import/first': 'error',
  'import/group-exports': 'off',
  'import/max-dependencies': 'off',
  'import/named': 'error',
  'import/namespace': 'error',
  'import/newline-after-import': 'error',
  'import/no-absolute-path': [
    'error',
    { esmodule: true, commonjs: true, amd: false },
  ],
  'import/no-amd': 'error',
  'import/no-anonymous-default-export': 'error',
  'import/no-commonjs': 'off',
  'import/no-cycle': 'warn',
  'import/no-default-export': 'off',
  'import/no-deprecated': 'warn',
  'import/no-duplicates': 'off',
  'import/no-dynamic-require': 'off',
  'import/no-empty-named-blocks': 'off', // revisit
  'import/no-extraneous-dependencies': 'error',
  'import/no-import-module-exports': 'error',
  'import/no-internal-modules': 'off', // revisit
  'import/no-mutable-exports': 'error',
  'import/no-named-as-default': 'error',
  'import/no-named-as-default-member': 'error',
  'import/no-named-default': 'error',
  'import/no-named-export': 'off',
  'import/no-namespace': 'off',
  'import/no-nodejs-modules': 'off',
  'import/no-relative-packages': 'error',
  'import/no-relative-parent-imports': 'off',
  'import/no-restricted-paths': 'off',
  'import/no-self-import': 'error',
  'import/no-unassigned-import': 'off',
  'import/no-unresolved': ['error', { commonjs: true, caseSensitive: true }],
  'import/no-unused-modules': 'off',
  'import/no-useless-path-segments': 'error',
  'import/no-webpack-loader-syntax': 'error',
  'import/order': [
    'error',
    {
      'groups': [
        'builtin',
        'external',
        'type',
        'internal',
        ['parent', 'sibling', 'index'],
      ],
      'newlines-between': 'always',
      'alphabetize': {
        order: 'asc',
        caseInsensitive: true,
      },
    },
  ],
  'import/prefer-default-export': 'off',
  'import/unambiguous': 'off',
}
