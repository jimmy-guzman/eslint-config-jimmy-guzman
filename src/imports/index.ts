import { rules } from './rules'

/**
 * https://github.com/import-js/eslint-plugin-import/tree/v2.25.2#rules
 */
export = {
  plugins: ['import'],
  rules,
  overrides: [
    {
      files: ['**/?(*.)+(stories).[jt]s?(x)'],
      rules: {
        'import/exports-last': 'off',
      },
    },
  ],
  settings: {
    'import/extensions': ['.js', '.mjs', '.jsx', '.ts', '.tsx'],
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx', '.json'],
      },
    },
    'import/external-module-folders': ['node_modules', 'node_modules/@types'],
    'import/ignore': ['node_modules'],
  },
}
