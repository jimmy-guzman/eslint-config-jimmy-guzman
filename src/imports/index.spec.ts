import imports from './index'

describe('imports', () => {
  it('should export imports rules config', () => {
    expect(Object.keys(imports.rules)).toHaveLength(44)
    expect(imports).toMatchSnapshot()
  })
})
