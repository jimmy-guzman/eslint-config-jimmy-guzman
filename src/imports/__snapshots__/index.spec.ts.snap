// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports should export imports rules config 1`] = `
Object {
  "plugins": Array [
    "import",
  ],
  "rules": Object {
    "import/default": "error",
    "import/dynamic-import-chunkname": "off",
    "import/export": "error",
    "import/exports-last": "error",
    "import/extensions": Array [
      "error",
      "ignorePackages",
      Object {
        "js": "never",
        "jsx": "never",
        "mjs": "never",
        "ts": "never",
        "tsx": "never",
      },
    ],
    "import/first": "error",
    "import/group-exports": "off",
    "import/max-dependencies": "off",
    "import/named": "error",
    "import/namespace": "error",
    "import/newline-after-import": "error",
    "import/no-absolute-path": Array [
      "error",
      Object {
        "amd": false,
        "commonjs": true,
        "esmodule": true,
      },
    ],
    "import/no-amd": "error",
    "import/no-anonymous-default-export": "error",
    "import/no-commonjs": "off",
    "import/no-cycle": "warn",
    "import/no-default-export": "off",
    "import/no-deprecated": "warn",
    "import/no-duplicates": "off",
    "import/no-dynamic-require": "off",
    "import/no-extraneous-dependencies": "error",
    "import/no-import-module-exports": "error",
    "import/no-internal-modules": "off",
    "import/no-mutable-exports": "error",
    "import/no-named-as-default": "error",
    "import/no-named-as-default-member": "error",
    "import/no-named-default": "error",
    "import/no-named-export": "off",
    "import/no-namespace": "off",
    "import/no-nodejs-modules": "off",
    "import/no-relative-packages": "error",
    "import/no-relative-parent-imports": "off",
    "import/no-restricted-paths": "off",
    "import/no-self-import": "error",
    "import/no-unassigned-import": "off",
    "import/no-unresolved": Array [
      "error",
      Object {
        "caseSensitive": true,
        "commonjs": true,
      },
    ],
    "import/no-unused-modules": "off",
    "import/no-useless-path-segments": "error",
    "import/no-webpack-loader-syntax": "error",
    "import/order": Array [
      "error",
      Object {
        "alphabetize": Object {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": Array [
          "builtin",
          "external",
          "type",
          "internal",
          Array [
            "parent",
            "sibling",
            "index",
          ],
        ],
        "newlines-between": "always",
      },
    ],
    "import/prefer-default-export": "off",
    "import/unambiguous": "off",
  },
  "settings": Object {
    "import/extensions": Array [
      ".js",
      ".mjs",
      ".jsx",
      ".ts",
      ".tsx",
    ],
    "import/external-module-folders": Array [
      "node_modules",
      "node_modules/@types",
    ],
    "import/ignore": Array [
      "node_modules",
    ],
    "import/resolver": Object {
      "node": Object {
        "extensions": Array [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
          ".json",
        ],
      },
    },
  },
}
`;
